sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -561,10 +561,11 @@ namespace sycl::ext::oneapi::experimental {
 template <typename T, typename PropertyListT = properties<>>
 class annotated_ref {
   public:
+    annotated_ref(T*);
```

Comment:
Is this part of the API, or is it just an implementation detail?  If the only way to get an `annotated_ref` is from `annotated_ptr`, this should not be part of the API.  Instead, it should be a private constructor and `annotated_ptr` should be a friend.